dss-xades/src/main/java/eu/europa/esig/dss/xades/XAdESIdPrefixes.java

```diff
@@ -0,0 +1,141 @@
+/**
+ * DSS - Digital Signature Services
+ * Copyright (C) 2015 European Commission, provided under the CEF programme
+ *
+ * This file is part of the "DSS - Digital Signature Services" project.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package eu.europa.esig.dss.xades;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class XAdESIdPrefixes {
+	private static final String ID_PREFIX = "id-";
+	private static final String TIMESTAMP_PREFIX = "ts-";
+	private static final String ENCAPSULATED_TIMESTAMP_PREFIX = "ets-";
+	private static final String KEYINFO_PREFIX = "keyInfo-";
+	private static final String VALUE_PREFIX = "value-";
+	private static final String XADES_PREFIX = "xades-";
+
+	/** Xml Id prefix */
+	private final String id;
+	/** Id-prefix for TimeStamp element */
+	private final String timestamp;
+	/** Id-prefix for EncapsulatedTimeStamp element */
+	private final String encapsulatedTimestamp;
+	/** Id-prefix for KeyInfo element */
+	private final String keyInfo;
+	/** Id-prefix for SignatureValue element */
+	private final String value;
+	/** Id-prefix for Signature element */
+	private final String xades;
+
+	public String getId() {
+		return id;
+	}
+
+	public String getTimestamp() {
+		return timestamp;
+	}
+
+	public String getEncapsulatedTimestamp() {
+		return encapsulatedTimestamp;
+	}
+
+	public String getKeyInfo() {
+		return keyInfo;
+	}
+
+	public String getValue() {
+		return value;
+	}
+
+	public String getXades() {
+		return xades;
+	}
+
+	public static XAdESIdPrefixes.XAdESIdBuilder newBuilder() {
+		return new XAdESIdPrefixes.XAdESIdBuilder();
+	}
+
+	private XAdESIdPrefixes(XAdESIdBuilder builder) {
+		this.id = builder.id;
+		this.timestamp = builder.timestamp;
+		this.encapsulatedTimestamp = builder.encapsulatedTimestamp;
+		this.keyInfo = builder.keyInfo;
+		this.value = builder.value;
+		this.xades = builder.xades;
+		validate();
+	}
+
+	private void validate() {
+		Set<String> attributeSet = new HashSet<>();
+		attributeSet.add(id);
+		attributeSet.add(timestamp);
+		attributeSet.add(encapsulatedTimestamp);
+		attributeSet.add(keyInfo);
+		attributeSet.add(value);
+		attributeSet.add(xades);
+
+		if (attributeSet.size() != 6) {
```

Comment:
I think we don't need to enforce uniqueness of the prefix. The id itself will be unique anyway and it doesn't serve our use case either.

Reply:
I see, but this test verifies on the entire id value, not only the prefix. Since UUID is part of the id, the id:s should be unique even when the same prefix is used.

Reply:
Unfortunately they have code that generates the same id if the prefixes are not unique